src/orion/testing/algo.py

```diff
@@ -172,54 +174,80 @@ def set_phases(cls, phases):
+        space: Space | None = None,
+        seed: int | Sequence[int] | None = None,
+        **kwargs,
+    ) -> SpaceTransformAlgoWrapper[AlgoType]:
+        """Create the algorithm based on config.
 
-        for method_name in sorted(cls_methods_with_phase):
-            parametrize_this(cls, method_name, attrs, ids)
+        Also initializes the algorithm with the required number of random trials from the previous
+        test phases before returning it.
 
-    def create_algo(self, config=None, space=None, **kwargs):
-        """Create the algorithm based on config.
+        If `seed` is passed, then `seed_rng` is called before observing anything.
 
         Parameters
         ----------
         config: dict, optional
-            The configuration for the algorithm. ``self.config`` will be used
+            The configuration for the algorithm. ``cls.config`` will be used
             if ``config`` is ``None``.
         space: ``orion.algo.space.Space``, optional
-            Space object to pass to algo. The output of ``self.create_space()``
+            Space object to pass to algo. The output of ``cls.create_space()``
             will be used if ``space`` is ``None``.
         kwargs: dict
             Values to override algorithm configuration.
         """
-        config = copy.deepcopy(config or self.config)
+        config = copy.deepcopy(config or cls.config)
         config.update(kwargs)
-        base_algo_type = orion.algo.base.algo_factory.get_class(self.algo_name)
-        original_space = space or self.create_space()
-        algo = create_algo(space=original_space, algo_type=base_algo_type, **config)
-        algo.algorithm.max_trials = self.max_trials
+
+        original_space = space or cls.create_space()
+        algo = create_algo(space=original_space, algo_type=cls.algo_type, **config)
+        # TODO: Add a `max_trials` attribute on the BaseAlgorithm class.
+        algo.algorithm.max_trials = cls.max_trials
+
+        # Seed the randomness before we observe anything.
+        if seed is not None:
+            algo.seed_rng(seed)
+
+        if cls._current_phase is not None and cls._current_phase.n_trials > 0:
+            n_previous_trials = cls._current_phase.n_trials
+            if n_previous_trials >= cls.max_trials:
+                raise ValueError(
+                    f"Test isn't configured properly: max_trials ({cls.max_trials}) is larger than "
+                    f"the total number of trials seen so far when in phase ({cls._current_phase}). "
+                    f"Increasing max_trials might be a good idea. "
+                )
+
+            # Force the algo to observe the given number of trials.
+            cls.force_observe(num=n_previous_trials, algo=algo)
+            # TODO: Should we check that the algo has indeed observed the right number of trials that we
+            # want, and that the max_trials hasn't been busted, if present?
+            assert algo.n_observed == n_previous_trials
+
```

Comment:
So far the algorithms are not cached to be reused across the tests for a given phase right?

Reply:
Yeah, they aren't. There would be a problem with the ordering of the parameterization from the fixture. I think I left a note somewhere about this.
I'll remove this TODO, it's outdated anyhow.

Reply:
Removed the todo in e456637ec38ea396be7f1f3713c52dbe7e65d63d